memfs/file.go

```diff
@@ -146,6 +146,11 @@ func (f *file) Unlock() error {
 	return nil
 }
 
+// Sync is a no-op in memfs.
+func (f *file) Sync() error {
```

Comment:
To avoid LSP violations, we don't implement the Sync on any structure that does not really support it.